project/products/idea/src/net/ashald/envfile/products/idea/IdeaRunConfigurationExtension.java

```diff
@@ -50,7 +53,14 @@ protected void validateConfiguration(@NotNull RunConfigurationBase configuration
 
     @Override
     public <T extends RunConfigurationBase> void updateJavaParameters(T configuration, JavaParameters params, RunnerSettings runnerSettings) throws ExecutionException {
-        EnvFileConfigurationEditor.patchEnvironmentVariables(configuration, params.getEnv());
+        Map<String, String> newEnv = new HashMap<>();
```

Comment:
This is a common thing that we will have to do in all supported products to it's better to minimize number of actions we perform here. I think it'd be better to do it as:
```
Map<String, String> newEnv = EnvFileConfigurationEditor.patchEnvironmentVariables(configuration, params.getEnv());
params.setEnv(newEnv);
```
so that `EnvFileConfigurationEditor.patchEnvironmentVariables` would be taking care of producing the overall environment variables set but would rather return it as a new map rather than modify an existing one in-place. This would minimize changes as well.

Reply:
Thanks for suggestion, I'll do it. 
As each product provides it's own API for interaction with environment variables the part of updating/replacing them will be different in each extension though.

Reply:
There is a common interface that we can rely on - each of them provides access to a `RunConfigurationBase` instance (that we can use to read env files) and a map of environment variables that were set previously. So I don't think that should be an issue.

Reply:
As I found out the actual outlier is the IDEA extension that doesn't call `RunConfigurationExtensionBase#patchCommandLine` that could have been the unified place of updating environment variables.

Please have a look at the new commits I added.